packages/backend/server/src/core/auth/guard.ts

```diff
@@ -60,70 +52,39 @@ export class AuthGuard implements CanActivate, OnModuleInit {
       return true;
     }
 
-    if (!req.user) {
+    if (!userSession) {
       throw new AuthenticationRequired();
     }
+
     return true;
   }
 
-  async signIn(
-    req: Request
-  ): Promise<{ user: CurrentUser; session: UserSession } | null> {
-    if (req.user && req.session) {
-      return {
-        user: req.user,
-        session: req.session,
-      };
+  async signIn(req: Request): Promise<Session | null> {
+    if (req.session) {
+      req.session;
```

Comment:
missing return

Reply:
prefer return with 304 code

Reply:
all of them are API, redirect is useless actually